### ISSUE TITLE
Keyword stuffing heuristic false-positive on natural-language descriptions with inline lists

### ISSUE THREAD

**Issue by dacharyc** [agent-ecosystem/skill-validator#26]
## Summary

The comma-separated-segments keyword stuffing heuristic in `checkDescriptionKeywordStuffing` (`structure/frontmatter.go`) fires a false positive on well-written, natural-language descriptions that happen to contain inline enumeration lists within sentences.

## Reproduction

The following description from [mongodb/agent-skills#10](https://github.com/mongodb/agent-skills/pull/10) causes the validator to fail CI:

```
Manages MongoDB Atlas Stream Processing (ASP) workflows. Handles workspace provisioning, data source/sink connections, processor lifecycle operations, debugging diagnostics, and tier sizing. Supports Kafka, Atlas clusters, S3, HTTPS, and Lambda integrations for streaming data workloads and event processing. NOT for general MongoDB queries or Atlas cluster management. Requires MongoDB MCP Server with Atlas API credentials.
```

This triggers:

> ⚠️ description has 9 comma-separated segments, most very short — this looks like a keyword list; per the spec, the description should concisely describe what the skill does and when to use it

## Root cause

Heuristic 2 in `checkDescriptionKeywordStuffing` splits the entire description on `,` and checks whether ≥60% of resulting segments are ≤3 words. This works for actual keyword lists like `"MongoDB, Atlas, Vector Search, embeddings, RAG, ..."` but doesn't distinguish inline comma-separated items within complete sentences from bare keyword lists.

For the description above, splitting on commas produces 9 segments:

| # | Segment (trimmed) | Word count | Short? |
|---|---|---|---|
| 1 | `Manages MongoDB Atlas Stream Processing (ASP) workflows. Handles workspace provisioning` | 8 | No |
| 2 | `data source/sink connections` | 3 | **Yes** |
| 3 | `processor lifecycle operations` | 3 | **Yes** |
| 4 | `debugging diagnostics` | 2 | **Yes** |
| 5 | `and tier sizing. Supports Kafka` | 5 | No |
| 6 | `Atlas clusters` | 2 | **Yes** |
| 7 | `S3` | 1 | **Yes** |
| 8 | `HTTPS` | 1 | **Yes** |
| 9 | `and Lambda integrations for streaming data workloads and event processing. NOT for general MongoDB queries or Atlas cluster management. Requires MongoDB MCP Server with Atlas API credentials.` | 24 | No |

6/9 = 67% short → triggers the ≥60% threshold. But this description is five complete, well-formed sentences — not a keyword list.

## Suggested fix directions

A few approaches that could reduce false positives without losing the ability to detect actual keyword stuffing:

1. **Sentence-aware splitting**: Only apply the comma-segment heuristic within individual sentences (split on `.` or `!` or `?` first). The description above has 5 sentences; none of them individually have ≥8 comma-separated segments, so it would pass.

2. **Raise the segment threshold**: The current threshold of 8 segments is quite low for descriptions that cover a multi-capability tool. Raising it (e.g., to 12) would avoid flagging descriptions with natural inline lists while still catching pure keyword dumps.

3. **Check for sentence structure signals**: If a majority of segments are preceded by verbs, articles, or conjunctions (`Handles`, `Supports`, `and`), that's a strong signal of prose rather than keyword stuffing. The heuristic could check for sentence-like markers and skip the warning.

4. **Raise the short-segment percentage threshold**: 60% is aggressive. Many natural English sentences with 4+ list items will produce ≥60% short segments. Raising this to 75% or 80% would be more precise.

Any combination of these would help. Option 1 seems like the most targeted fix.
